### PR TITLE
Sheep Shearer Rewrite

### PR DESCRIPTION
**Pull request by Obasill**
More of a walk through and slightly more elegant solution. With steps on how to get the wool, turn wool into balls of wool, and then turn in. It's not perfect but I added things to catch edge cases such as opening the helper mid-quest progress.

### CODE REVIEW

**Review by Zoinkwiz — CHANGES_REQUESTED** [review]
Currently this is a bit problematic for if a player has a semi-full inventory, as the quest won't progress until you have all the wool. Also, if a player hands in partial quantities it'll cause the helper to close as steps don't exist for the intermediate steps.

File: src/main/java/com/questhelper/helpers/quests/sheepshearer/SheepShearer.java [hunk 1]
```diff
@@ -44,110 +59,136 @@
...
+		if (inventory == null) return;
+
+		int itemsInInventory = inventory.count();
+		skipIfFullInventory.setShouldPass(itemsInInventory == 28);
+	}
+
+	public void setupSteps()
+	{
+		startStep = new NpcStep(this, NpcID.FRED_THE_FARMER, new WorldPoint(3190, 3273, 0),
+			"Talk Fred the Farmer north of Lumbridge to start the quest. Bring 20 balls of wool to autocomplete the quest.");
```
Comment (pajlada) [inline src/main/java/com/questhelper/helpers/quests/sheepshearer/SheepShearer.java:147]:
```suggestion
			"Talk to Fred the Farmer north of Lumbridge to start the quest. Bring 20 balls of wool to autocomplete the quest.");
```

File: src/main/java/com/questhelper/helpers/quests/sheepshearer/SheepShearer.java [hunk 2]
```diff
@@ -44,110 +59,136 @@
...
 	}
 
 	@Override
 	public void setupRequirements()
 	{
-		twentyBallsOfWool = new ItemRequirement("Balls of wool", ItemID.BALL_OF_WOOL, 20);
-		shears = new ItemRequirement("Shears if you plan on collecting wool yourself", ItemID.SHEARS).isNotConsumed();
+		ballOfWool = new ItemRequirement("Balls of wool", ItemID.BALL_OF_WOOL);
+		shears = new ItemRequirement("Shears", ItemID.SHEARS).isNotConsumed();
+		shears.setTooltip("if you plan on collecting wool yourself");
```
Comment (pajlada) [inline src/main/java/com/questhelper/helpers/quests/sheepshearer/SheepShearer.java:117]: nit: capitalization
```suggestion
		shears.setTooltip("If you plan on collecting wool yourself");
```

**Review by pajlada — CHANGES_REQUESTED** [review]
Additionally, when going for my last sheep to shear, quest helper didn't prompt me to spin them into balls of wool even though I should have at this point
image

Not fully related to this PR, but I noticed the Sheep don't get highlighted again after growing their wool back

**Pull request merged**